pimd/pim_pim.c

```diff
@@ -596,13 +596,24 @@ static int pim_msg_send_frame(int fd, char *buf, size_t len,
 }
 
 int pim_msg_send(int fd, pim_addr src, pim_addr dst, uint8_t *pim_msg,
-		 int pim_msg_size, const char *ifname)
+		 int pim_msg_size, struct interface *ifp)
 {
 	socklen_t tolen;
 	unsigned char buffer[10000];
 	unsigned char *msg_start;
 	uint8_t ttl;
 	struct pim_msg_header *header;
+	struct pim_interface *pim_ifp;
+
+	pim_ifp = ifp->info;
+
+	if (PIM_IF_TEST_PASSIVE_ENABLE(pim_ifp->options)) {
+		if (PIM_DEBUG_PIM_PACKETS)
+			zlog_debug(
+				"could not send PIM message on passive interface %s",
+				ifp->name);
+		return 0;
+	}
```

Comment:
Misleading debug message. We could send but chose not to. If we want to log this it should be like :

"skip sending PIM message on passive interface %s".

Since we are not even logging the message type, not sure if  it is that useful, but it doesn't hurt.

Reply:
Done